pkg/controllers/user/rbac/legacy_grb_cleaner.go

```diff
@@ -19,27 +19,32 @@ func newLegacyGRBCleaner(m *manager) *grbCleaner {
 	return &grbCleaner{m: m}
 }
 
+// this function addresses issues with grb not being cleaned up that was an issue from v2.2.3 - v2.2.8
+// it works on previously created objects outside of the timeline of normal sync handlers to correct issues with finalizers not be removed on deletion and finalizers not being dropped on cluster deletion
 func (p *grbCleaner) sync(key string, obj *v3.GlobalRoleBinding) (runtime.Object, error) {
 	if key == "" || obj == nil {
 		return nil, nil
 	}
-	if obj.DeletionTimestamp == nil || len(obj.Finalizers) == 0 {
-		return nil, nil
-	}
-	if time.Since(obj.DeletionTimestamp.Time) < 1*time.Hour {
-		return nil, nil
+	if obj.Annotations[grbstore.GrbVersion] == "true" {
+		return obj, nil
 	}
-	obj, err := p.removeStuckFinalizer(obj)
+	obj, err := p.removeFinalizerFromNonExistentCluster(obj)
 	if err != nil && !errors.IsNotFound(err) {
-		return nil, err
+		return obj, err
 	}
-	return obj, nil
-
+	annotations := obj.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
+		obj.SetAnnotations(annotations)
+	}
+	obj.Annotations[grbstore.GrbVersion] = "true"
+	return p.m.workload.Management.Management.GlobalRoleBindings("").Update(obj)
 }
 
-func (p *grbCleaner) removeStuckFinalizer(obj *v3.GlobalRoleBinding) (*v3.GlobalRoleBinding, error) {
+func (p *grbCleaner) removeFinalizerFromNonExistentCluster(obj *v3.GlobalRoleBinding) (*v3.GlobalRoleBinding, error) {
 	obj = obj.DeepCopy()
 	modified := false
```

Comment:
Drop the modified check, just settings the finalizers regardless is fine and would be a no-op anyways if no changes happen.